api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -203,15 +209,16 @@ func (i *MaterializeStageProvider) Process(ctx context.Context, mgrContext conte
 			metrics.RunOperationType,
 			v1alpha2.CatalogsGetFailed.String(),
 		)
-		if anyCatalogNotExist {
+		if anyCatalogInvalid {
```

Comment:
I think when the length is different, should always throw `BadRequest` as customer use more catalogs to materials using input. What do you think? Do we need the notFound check?

Reply:
If GetCatalog fails with transient issue like binding unavailable, we need to show InternalError rather than badRequest

Reply:
One more suggestion is we'd better change the error code filtering. We'd better not use error string contains. @RemindD , please create an issue to improve this.